attestation-agent/attester/src/snp/mod.rs

```diff
@@ -25,17 +25,15 @@ struct SnpEvidence {
 pub struct SnpAttester {}
 
 impl Attester for SnpAttester {
-    fn get_evidence(&self, report_data: String) -> Result<String> {
-        let mut report_data_bin = base64::decode(report_data)?;
-
-        if report_data_bin.len() != 48 {
-            bail!("Malformed SNP Evidence");
+    fn get_evidence(&self, mut report_data: Vec<u8>) -> Result<String> {
+        if report_data.len() > 64 {
+            bail!("SNP Attester: Report data must be no more than 64 bytes");
         }
 
-        report_data_bin.extend([0; 16]);
+        report_data.resize_with(64, || 0);
```

Comment:
`report_data.resize(64, 0)` (if we want to keep the vec in the signature, we can just call)

Reply:
Nice suggestion! Let me fix this